maven-core/src/main/java/org/apache/maven/toolchain/DefaultToolchainManager.java

```diff
@@ -43,12 +46,25 @@
 public class DefaultToolchainManager
     implements ToolchainManager
 {
-    @Inject
-    Logger logger;
+    protected final Logger logger;
 
     @Inject
     Map<String, ToolchainFactory> factories;
 
+    @Inject
+    public DefaultToolchainManager()
+    {
+        this( LoggerFactory.getLogger( DefaultToolchainManager.class ) );
+    }
+
+    /**
+     * Ctor for UT.
+     */
+    public DefaultToolchainManager( Logger logger )
```

Comment:
If it is for a unittest, can you remove `public`?

Reply:
fixed